packages/amplify-category-api/src/provider-utils/awscloudformation/cdk-stack-builder/apigw-stack-transform.ts

```diff
@@ -0,0 +1,216 @@
+      },
+      'ApiName',
+    );
+
+    this.resourceTemplateObj.addCfnOutput(
+      {
+        value: cdk.Fn.ref(this.resourceName),
+        description: 'API ID (prefix of API URL)',
+      },
+      'ApiId',
+    );
+
+    // Add resources
+    this.resourceName === 'AdminQueries'
+      ? this.resourceTemplateObj.generateAdminQueriesStack(this.resourceName, authResourceName)
+      : this.resourceTemplateObj.generateStackResources(this.resourceName);
+  }
+
+  async applyOverrides() {
+    const backendDir = pathManager.getBackendDirPath();
+    const overrideFilePath = pathManager.getResourceDirectoryPath(undefined, AmplifyCategories.API, this.resourceName);
+    const overrideJSFilePath = path.join(overrideFilePath, 'build', 'override.js');
+
+    const isBuild = await buildOverrideDir(backendDir, overrideFilePath).catch(error => {
+      printer.debug(`Skipping build as ${error.message}`);
+      return false;
+    });
+
+    // skip if packageManager or override.ts not found
+    if (isBuild) {
+      const { override } = await import(overrideJSFilePath).catch(error => {
+        formatter.list(['No override file found', `To override ${this.resourceName} run "amplify override api"`]);
+        return undefined;
+      });
+
+      if (override && typeof override === 'function') {
+        const overrideCode: string = await fs.readFile(overrideJSFilePath, 'utf-8').catch(() => {
+          formatter.list(['No override File Found', `To override ${this.resourceName} run amplify override auth`]);
+          return '';
+        });
+
+        const sandboxNode = new vm.NodeVM({
+          console: 'inherit',
+          timeout: 5000,
+          sandbox: {},
+          require: {
+            context: 'sandbox',
+            builtin: ['path'],
+            external: true,
+          },
+        });
+
+        try {
+          await sandboxNode.run(overrideCode, overrideJSFilePath).override(this.resourceTemplateObj as AmplifyApigwResourceStack);
+        } catch (err: $TSAny) {
+          const error = new Error(`Skipping override due to ${err}.`);
+          printer.error(`${error}`);
+          error.stack = undefined;
+          throw error;
+        }
+      }
+    }
+  }
+
+  private async saveBuildFiles() {
+    if (this.resourceTemplateObj) {
+      this.cfn = JSONUtilities.parse(this.resourceTemplateObj.renderCloudFormationTemplate());
+    }
+
+    const resourceDirPath = pathManager.getResourceDirectoryPath(undefined, category, this.resourceName);
+    fs.ensureDirSync(resourceDirPath);
+
+    const buildDirPath = path.join(resourceDirPath, PathConstants.BuildDirName);
+    fs.ensureDirSync(buildDirPath);
+
+    stateManager.setResourceParametersJson(undefined, AmplifyCategories.API, this.resourceName, this.cfnInputParams);
+
+    const cfnFilePath = path.resolve(path.join(buildDirPath, `${this.resourceName}-cloudformation-template.json`));
+    return writeCFNTemplate(this.cfn, cfnFilePath);
```

Comment:
```suggestion
    return await writeCFNTemplate(this.cfn, cfnFilePath);
```

Reply:
`await` here isn't needed because we're returning a promise, so the caller will handle the awaiting.

Reply:
`await` here isn't needed because the function is returning a promise, so the caller will handle the awaiting.